openedx/frontend-app-authoring
# Let Studio Home REST API determine if libraries v1 and/or v2 are enabled [FC-0062]

## Description

Removes the `LIBRARY_MODE` config variable in favor of flags returned by the Studio Home REST API. See https://github.com/openedx/edx-platform/pull/35576 for the new waffle flags added there.

Also adds a "Beta" badge and explanatory text to the Libraries v2 home page.

This change impacts Course/Library Authors.

### Screenshots

* Both flags disabled (default, v1 + v2 libraries enabled):
  ![image](https://github.com/user-attachments/assets/d1a52b2f-c8c9-4df2-96f5-c120d9385cfb)
  
* Only `contentstore.new_studio_mfe.disable_new_libraries` flag enabled (v2 libraries disabled):
  ![image](https://github.com/user-attachments/assets/7d0d530c-c934-49d8-b912-5fe4f60dca68)
  
* Only `contentstore.new_studio_mfe.disable_legacy_libraries` enabled (v1 libraries disabled):
  ![image](https://github.com/user-attachments/assets/049d9947-6791-487e-bc73-5f8594bad2a4)
  
* Both flags enabled (both v1 + v2 libraries disabled):
  ![image](https://github.com/user-attachments/assets/7a60bdda-cb1d-4738-a0a3-06f7672cf7c1)


## Supporting information

Relates to:
* https://github.com/openedx/frontend-app-authoring/issues/1336
* https://github.com/openedx/frontend-app-authoring/issues/1324

Blocked by:
* https://github.com/openedx/edx-platform/issues/32457

Private-ref: [FAL-3858](https://tasks.opencraft.com/browse/FAL-3858)

## Testing instructions

1. Run branch from https://github.com/openedx/edx-platform/issues/32457, 
   and ensure the Meilisearch plugin is enabled: https://github.com/open-craft/tutor-contrib-meilisearch
2. View Authoring Home page: should show "Libraries (beta)" tab linked to `/libraries` and "Legacy Libraries" tab linked to `/libraries-v1` 
3. In the Django Admin, add a Waffle Flag named `contentstore.new_studio_mfe.disable_new_libraries` and enable it for everyone. Refresh Authoring home page
   The "Libraries (beta)" tab should be gone now, and "Legacy Libraries" should be titled "Libraries", still linking to `/libraries-v1`
4. In the Django Admin, add a Waffle Flag named `contentstore.new_studio_mfe.disable_legacy_libraries` and enable it for everyone. Refresh Authoring home page
   No "Libraries" tabs should be shown.
5. In Django Admin, disable `contentstore.new_studio_mfe.disable_new_libraries` for everyone.
   The "Libraries (beta)" tab should be the only tab showing, linked to `/libraries`.
6. In Django Admin, disable both libraries flags.
   Should be back to the default view with both "Libraries (beta)" and "Legacy Libraries" showing.

## Other information

1. The link in the explanatory text will be updated once we have a tutorial available.
